tshirt page changes for review (#5364)
grommet/hpe-design-system: aries-site/src/pages/foundation/tshirt-sizing.mdx

import { Example } from '../../layouts';
import { ButtonSizingExample } from '../../examples';

## Summary

- Named size tokens (e.g. `small`, `medium`) replace raw pixel values.
- The system governs both component dimensions and spacing (padding, gap, margin, border, radius).

> - The scale derives from two base units: 24px (original) and 16px (supplemental in v7).

Reviewer: We jump in to talking about v7 and v6. Since we have a lot of different package offering s let's make it clear that we are referring to grommet-theme-hpe here.

Let's also try to structure the information in such a way that it will hold up long term. For example if we are on v9 or v10 of the theme, this page should still make sense to users. The way that this reads currently is that the supplemental base unit only applys to v7

Finally, t-shirt sizes enable content to responsively scale more easily and consistently. For example, a layout's padding may be set to "large" on desktop screen and gracefully adjust to "small" on mobile devices with minimal effort.

## The dual doubling scale (HPE theme v7 and above)

With the introduction of HPE theme v7, the t-shirt sizing scale has been significantly expanded using a "Dual Doubling" approach to provide more granular sizing options. This new scale addresses gaps in the previous scale and offers greater design flexibility.

**Key highlights of the Dual Doubling scale:**

- **Introduces in-between steps** which address gaps in the current scale, making desired design layouts easier to execute.

> - **Has parity with existing scale** which means that all designs may be recreated with the new scale without any layout shifts except for example container `xlarge` or 1152px doesn't have an equivalent size on the new scale. We recommend using `xxlarge` or 1024px instead on the new scale.

Reviewer 2: ```suggestion
- **Has parity with existing scale** which means that all designs may be recreated with the new scale without any layout shifts. The only exception is container `xlarge` (1152px), which does not have a direct equivalent. We recommend using `xxlarge` (1024px) instead.
```

- **New `medium` = v6 and below `medium`**; however, rest of t-shirt sizes will need to be mapped to achieve design parity.

> - **Every t-shirt size may be composed** by smaller t-shirt sizes and/or divided into halves, thirds, quarters, ... 12ths, etc. (e.g. `small` + `small` = `large`, or `xsmall` + `xsmall` + `xsmall` = `large`, or even `medium` + `xsmall` = `large`).

Reviewer 2: **T-shirt sizes are modular:** ....

The [Grommet site](https://v2.grommet.io/) provides documentation on the props available to any given component. If you're unsure if a component supports `width`, `height`, or `size`, that would be the right place to check.

> Given the [Container sizes comparison](#container-sizes-comparison-width-height-size-columns-rows) table, if you were to implement the following code, you would have a first Box with width of "384px" and height of "192px", second Box with width of "384px" and height of "384px" and a third Box with width of "384px" and height of "768px"

Reviewer 2: ```suggestion
Given the [Container sizes comparison](#container-sizes-comparison-width-height-size-columns-rows) table the following code would render 3 boxes

- The first Box has a width of 384px and height of 192px.
- The second Box has a width of 384px and height of 384px.
- The third Box has a width of 384px and height of 768px.
```

not sure about this suggestion or how this helps user open to other thoughts

```
<Box width="medium" height="xsmall" />
// width = 384px, height = 192px

<Box width="medium" height="medium" />
// width = 384px, height = 384px

<Box width="medium" height="xlarge" />
// width = 384px, height = 768px
```

<Box width="xlarge">
  <Image src="/box-sizing.png" alt="Demonstration of t-shirt sizing on Box" />
</Box>

## T-shirt sizing for spacing and other styles

In addition to overall component dimensions, t-shirt sizing can also be used to define spacing and other styling such as pad, gap, margin, and border.

To create the pixel values for these styles, Grommet divides and/or multiplies the base unit of `24px` by various integers. The HPE theme v7 introduces additional scale doubling of 16px base unit to achieve intermediate sizes for more precise spacing control.

> Pad, gap, and margin are controlled by `edgeSize` which now includes the expanded scale shown in the spacing sizes section above.

Reviewer 2: ```suggestion
`pad`, `gap`, and `margin` are controlled by `edgeSize` which now includes the expanded scale shown in the spacing sizes section above.
```

    alt="Demonstration of t-shirt sizing for pad on Box"
  />
</Box>

The following snippet produces a Box with width 384px, height 192px, and a 2px border.

```
<Box
    border={{ color: 'border', size: 'small' }}
    width="medium"
    height="xsmall"
/>
```

<Box width="medium" margin={{ bottom: 'xlarge' }}>
  <Image
    src="/border-sizing.png"
    alt="Demonstration of t-shirt sizing for border on Box"
  />
</Box>

### Composability

The v7 scale maintains additive and divisional consistency; sums or partitions of smaller tokens align precisely with larger counterparts.

```
// Container sizes maintain doubling relationships at key intervals:
xsmall (192px) + xsmall (192px) = medium (384px)
medium (384px) + medium (384px) = xlarge (768px)

// New intermediate sizes fill gaps:
small (256px) = medium (384px) - xxsmall (128px)
large (512px) = medium (384px) + xxsmall (128px)

// Spacing sizes follow similar mathematical relationships:
xsmall (12px) + xsmall (12px) = medium (24px)
medium (24px) + medium (24px) = xlarge (48px)
```

Intermediate tokens address previous gaps where `medium` and `large` formed an overly broad jump.

### Composition visualization

> The beauty of the Dual Doubling scale is its mathematical precision, allowing sizes to compose and decompose in predictable ways. The visualization below demonstrates how different t-shirt sizes relate to each other:

Reviewer: In some places we refer to the 'v7 scale' in others we say 'the Dual Doubling scale', thinking we should align to a consistent language.

I'm wondering if we should refer to the v6 and earlier scale as the 'legacy' t-shirt scale and v7 and above is just the t-shirt size scale.
